### PR TITLE
LPS-157149 Add Tests for Testray Environments- Administrator

### PR DESCRIPTION
**Pull request by joelrocha68**
[LPS-157149](https://issues.liferay.com/browse/LPS-157149)

### CODE REVIEW

**Comment by joelrocha68** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro [hunk 1]
```diff
@@ -101,17 +177,21 @@ definition {
...
 		Type(
 			locator1 = "TestrayManage#NEW_OPTION_NAME",
-			value1 = "${valueType}");
+			value1 = "${key_value}");
 
 		Select(
-			locator1 = "TestrayManage#SELECT_CATEGORY",
-			value1 = "${category}");
+			locator1 = "TestrayManage#SELECT_CATEGORIES_IN_OPTIONS",
+			value1 = "${categorie}");
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro:190]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase [hunk 2]
```diff
@@ -18,47 +18,93 @@ definition {
...
+			TestrayManage.addNewCategory(category = "Environments_2 Factor Category");
+		}
+
+		task ("Go to Options on Manage Sidebar and Add New Option") {
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Options");
+
+			TestrayManage.createNewOption(
+				categorie = "Environments_2 Factor Category",
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase:49]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase [hunk 3]
```diff
@@ -71,39 +117,113 @@ definition {
...
+		}
+
+		task ("Go to Options on Manage Sidebar and Add New Option") {
+			TestrayLayout.clickManageOnSidebar();
 
-		// TODO LPS-157326 AdminCanEditOptionChangingCategory pending implementation
+			TestrayManage.selectOptionsManage(itemValue = "Options");
 
+			TestrayManage.createNewOption(
+				categorie = "Environments_8 Category A",
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase:180]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase [hunk 4]
```diff
@@ -71,39 +117,113 @@ definition {
...
+				key_value = "Environments_8 Option");
+		}
+
+		task ("Check if you can edit the Option by changing the Category") {
+			TestrayManage.editNewOptionName(
+				categorie = "Environments_8 Category B",
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase:186]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase [hunk 5]
```diff
@@ -71,39 +117,113 @@ definition {
...
+			TestrayManage.addNewCategory(category = "Environments_7 Category");
+		}
+
+		task ("Create the new Options and then edit and state if the name has changed") {
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Options");
+
+			TestrayManage.createNewOption(
+				categorie = "Environments_7 Category",
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase:216]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro [hunk 6]
```diff
@@ -38,18 +40,32 @@ definition {
 		Click(locator1 = "TestrayManage#CLOSE_MODAL");
 	}
 
+	macro assertCategorieNotPresent {
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro:43]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

assertCategoryNotPresent

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro [hunk 7]
```diff
@@ -77,7 +93,43 @@ definition {
 			locator1 = "TestrayManage#NEW_OPTION_NAME",
 			value1 = "${key_value}");
 
+		Select(
+			locator1 = "TestrayManage#SELECT_CATEGORIES_IN_OPTIONS",
+			value1 = "${categorie}");
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro:98]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro [hunk 8]
```diff
@@ -90,6 +142,30 @@ definition {
 		MenuItem.clickNoError(menuItem = "Delete");
 	}
 
+	macro editCategoryOption {
+		Type(
+			locator1 = "TestrayManage#SELECT_CATEGORIES_IN_OPTIONS",
+			value1 = "${categorie}");
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro:148]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase [hunk 9]
```diff
@@ -18,47 +18,93 @@ definition {
...
+			TestrayManage.addNewCategory(category = "Environments_2 Category");
+		}
+
+		task ("Go to Options on Manage Sidebar, Add Options and assert the new option has been created") {
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Options");
+
+			TestrayManage.createNewOption(
+				categorie = "Environments_2 Category",
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase:101]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase [hunk 10]
```diff
@@ -71,39 +117,113 @@ definition {
...
+		}
+
+		task ("Check if you can edit the Option by changing the Category") {
+			TestrayManage.editNewOptionName(
+				categorie = "Environments_8 Category B",
+				key_value = "Environments_8 Option");
+
+			TestrayManage.assertNameCategoryInOptions(category = "Environments_8 Category B");
+
+			TestrayManage.assertCategorieNotPresent(key_title = "Environments_8 Category A");
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase:191]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase [hunk 11]
```diff
@@ -71,39 +117,113 @@ definition {
...
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Options");
+
+			TestrayManage.createNewOption(
+				categorie = "Environments_7 Category",
+				key_value = "Environments_7 Option");
+
+			TestrayManage.editNewOptionName(
+				categorie = "Environments_7 Category",
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase:220]: Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro [hunk 12]
```diff
@@ -77,7 +93,43 @@ definition {
...
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Delete");
+
+			AssertConfirm(value1 = "Are you sure you want to delete this item?");
+
+			Alert.viewSuccessMessage();
+
+			CommerceUtilities.checkSuccessAlertAppearsAndDisappears();
+		}
+	}
+
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro:122]: Can we have this macro to accept an specific category to be deleted (similar to deleteOption) and another macro to delete all categories?

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase [hunk 13]
```diff
@@ -18,47 +18,93 @@ definition {
...
+			TestrayManage.createNewOption(
+				categorie = "Environments_2 Factor Category",
+				key_value = "Environments_2 Factor Option");
+		}
+
+		task ("Delete Option, Go to Options and state if deleted") {
+			TestrayManage.deleteOption(nameOptions = "Environments_2 Factor Option");
 
-		// TODO LPS-157327 AdminCanAddAndDeleteFactorOption pending implementation
+			TestrayManage.deleteCategory();
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase:56]: Please check the comment below the deleteCategory macro and change this test case accordingly.

**Review by barrosdennys — COMMENTED** [review]
@joelrocha68 would you please take a look at the comments?

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro [hunk 14]
```diff
@@ -90,6 +128,46 @@ definition {
 		MenuItem.clickNoError(menuItem = "Delete");
 	}
 
+	macro deleteSpecificCategory {
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro:131]: please change the macro name to deleteCategory

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro [hunk 15]
```diff
@@ -75,9 +77,45 @@ definition {
...
+		Select(
+			locator1 = "TestrayManage#SELECT_CATEGORY_IN_OPTIONS",
+			value1 = "${category}");
 
 		Click(locator1 = "TestrayManage#SAVE_BUTTON");
+
+		CommerceUtilities.checkSuccessAlertAppearsAndDisappears();
+	}
+
+	macro deleteCategory {
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayManage.macro:91]: please change the macro name to deleteAllCategories

**Comment by barrosdennys** [comment]
ci:forward

**Comment by joelrocha68** [comment]
ci:forward

**Pull request closed without merging**